jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -367,6 +378,51 @@ def _parse_spectrum1d(app, file_obj, data_label=None, spectrum_viewer_reference_
     app.add_data_to_viewer(spectrum_viewer_reference_name, data_label)
 
 
+def _parse_ndarray(app, file_obj, data_label=None, data_type=None,
+                   flux_viewer_reference_name=None, spectrum_viewer_reference_name=None,
+                   uncert_viewer_reference_name=None):
+    if data_label is None:
+        data_label = app.return_data_label(file_obj)
+
+    if data_type is None:
+        data_type = 'flux'
+
+    # Cannot change axis to ensure roundtripping within Cubeviz.
+    # Axes must already be (x, y, z) at this point.
+    flux = file_obj
```

Comment:
@rosteen , I thought about this and to ensure there is less things we need to undo when you fix the axis ordering upstream, I decided that going with your suggestion is the best option. Thanks!